### ISSUE TITLE
Issue collection

### ISSUE THREAD

**Issue by kuangliu** [kuangliu/pytorch-retinanet#11]
Hey guy, I'm super busy the two weeks. Finally I get some time working on this.
For now, let's fix the issue one by one.

@njtuzzy:
>@kuangliu @Mendel1 In the encoder file, the output of "get_anchor_boxes" is the "xcenter,ycenter, xwidth, ywidth" format, it seems that it does not need to change to xxwh(**I guess you mean `xywh`**) using change_box_order function?

`anchor_boxes` is ordered as `xywh`
`boxes` is changed from `xyxy` to `xywh` with `chang_box_order`
```Python
boxes = change_box_order(boxes, 'xyxy2xywh')
```
Now they are both `xywh`. Any problems?

**Comment by c0nn3r** [comment]
I do you think you should look at my original list of issues, but I do understand too many comments were made.

**Comment by Mendel1** [comment]
@kuangliu  Can you upload your test result?Could your net  predict  correctly?

**Comment by kuangliu** [comment]
No. I know the loss is not working. 
I would like to know is there any bug in the `encoder` first.

**Comment by Mendel1** [comment]
I think decode is wrong.At least it's not the way the paper indicates.

**Comment by njtuzzy** [comment]
@kuangliu  yes  you are correct !

**Comment by kuangliu** [comment]
Cool. I update the loss. I'm still testing it. 
The loss vibrates. Not sure what's going wrong.

**Comment by njtuzzy** [comment]
@kuangliu may i ask the class label is from 0 to 20 for voc or 1 to 21??

**Comment by kuangliu** [comment]
Yeah. Just as Faster RCNN.

**Comment by njtuzzy** [comment]
@kuangliu  do you initalize the bias of final cov-layer  as section 4.1 mentioned?

**Comment by njtuzzy** [comment]
@kuangliu  In the paper, they said "anchors are assigned to GT object boxes usng IOU>0.5 and each anchor is assigned to at most one object boxes"? it seems you only take the biggest IOU in classes for each anchor? is there any difference?

**Comment by philokey** [comment]
@kuangliu the latest code will get nan in training processing.

**Comment by xueeinstein** [comment]
@kuangliu the loss explodes while training like this:

![image](https://user-images.githubusercontent.com/3424201/31317004-9b1e9000-ac6b-11e7-88c6-afd3f1494651.png)

If it works for you now, please publish a detection demo, if possible, with a available pytorch model. Thanks in advance!

**Comment by kobehuai** [comment]
the function encoder.decode is work? 
I try to run the test.py after training but find the score is null after  `score, labels = cls_preds.sigmoid().max(0)`, I guess the gt is encoded and the cls_preds and loc_preds don't need decode when training, so it works well. 
But something go wrong in the test process when the preds output need decode, are there something wrong in the  encoder.decode func?

**Comment by kobehuai** [comment]
the function encoder.decode is work?
I try to run the test.py after training but find the score is null after `score, labels = cls_preds.sigmoid().max(1)`, I guess the gt is encoded and the cls_preds and loc_preds don't need decode when training, so it works well.
But something go wrong in the test process when the preds output need decode, are there something wrong in the encoder.decode func?

**Comment by dami23** [comment]
@kuangliu Dear Kuangliu, where can I download the pretrained model metioned in the code? Could you give the address of the pretrained model? Thank you very much!!

**Comment by kehaozhe** [comment]
@kobehuai 
hi, have you solve this problems?  I am working with the same question.
thank you if you can share you solution.

**Comment by kobehuai** [comment]
@kehaozhe 
I find there are some nan samples in the cls_preds, so before the sigmoid, I set all the nan to -1:
`cls_preds[torch.isnan(cls_preds)] = -1`
But the fundamental problem is that the network should not output nan, I still have not solved this problem.
